packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1871,6 +1871,13 @@ def __init__(
         logger: logging.Logger,
     ):
         """Initialize the AbciApp."""
+
+        if not hasattr(self.initial_round_cls, "synchronized_data_class"):
+            logger.warning(f"No `synchronized_data_class` set on {self}")
+        else:
+            synchronized_data_class = self.initial_round_cls.synchronized_data_class  # type: ignore
+            synchronized_data = synchronized_data_class(db=synchronized_data.db)
+
```

Comment:
This is I believe the minimally invasive solution that allows casting to the correct type on initialization of the `AbciApp`. This does not yet deal with chained applications, which requires a separate test and solution only worth implementing if this solution is approved.

Reply:
Why do you think it needs a different implementation? 

It certainly needs a test.